nixos/modules/services/matrix/synapse.nix

```
@@ -1008,6 +1042,8 @@ in {
     # default them, so they are additive
     services.matrix-synapse.extras = defaultExtras;
 
+    services.matrix-synapse.log = defaultCommonLogConfig;
```

If we would use something like, then mkForce wouldn't be required.

```suggestion
    services.matrix-synapse.log = mapAttrsRecursive (_: lib.mkDefault) defaultCommonLogConfig;
```

---

Yeah, I considered doing something like that, but I'm not sure if that's too much magic (it would be less of an issue if we didn't nested attribute sets here IMHO).

What do the others think?

---

I think making this recursively default makes it a lot harder for people who want to override the log config in its entirety as every property would need to be unset/overridden. I'm not sure though whether there's actually a useful use case for this, so unless we find any I'm fine with the proposed change to `mkDefault`.

---

> I think making this recursively default makes it a lot harder for people who want to override the log config in its entirety as every property would need to be unset/overridden

This should be doable with `services.matrix-synapse.log = mkForce ...;`.

The sole difference of what @SuperSandro2000 suggested should be whether you need to override e.g. `services.matrix-synapse.log.root.level` with `mkForce` (currently the case with my patch) or not (doable when recursively doing `mkDefault`).

However, revisiting this again: I was concerned about corner cases (where it's more correct to `mkDefault` an attribute-set rather than its attribtues inside), but we know the default, i.e. the configuration where `mapAttrsRecursive` gets applied upon and that one seems fine, so I'm inclined to just do that.

---

I think I am fine with both options :)